### PR TITLE
Serve rostrum with Golang

### PR DESCRIPTION
**Pull request by nehayward**
This provides simple serving of rostrum frontend with go.

Also included is dockerfile and make to build and run.

close #23

### CODE REVIEW

File: rostrum/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM alpine
+ADD ./build /build
```
Comment (zpatrick) [inline rostrum/Dockerfile]: Do we need to add certs, ala [layer0](https://github.com/quintilesims/layer0/blob/develop/api/Dockerfile#L2)?

File: rostrum/Makefile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+SHELL:=/bin/bash
+VERSION?=$(shell git describe --tags --always)
+NAME:=rostrum
+CURRENT_DOCKER_IMAGE=quintilesims/$(NAME):$(VERSION)
+LATEST_DOCKER_IMAGE=quintilesims/$(NAME):latest
+
+build:
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a --ldflags "-X main.Version=$(VERSION)" -o $(NAME) .
```
Comment (zpatrick) [inline rostrum/Makefile]: we should build the react app here as well

File: rostrum/Makefile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	docker build -t $(CURRENT_DOCKER_IMAGE) .
+
+release: build
+	docker push $(CURRENT_DOCKER_IMAGE)
+	docker tag $(CURRENT_DOCKER_IMAGE) $(LATEST_DOCKER_IMAGE)
+	docker push $(LATEST_DOCKER_IMAGE)
+
+run:
+	docker run --rm -d -p 80:80 $(CURRENT_DOCKER_IMAGE)
```
Comment (zpatrick) [inline rostrum/Makefile]: We typically execute with `go run` when developing locally, given docker slows down local iteration speed quite significantly. 
Also, I think we should use something other than port 80 for local dev, given that's typically reserved/in use.

File: rostrum/main.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import (
+	"flag"
+	"log"
+	"net/http"
+)
+
+func main() {
+	var port string
+
+	flag.StringVar(&port, "port", "3000", "the `port` to listen on.")
```
Comment (zpatrick) [inline rostrum/main.go]: We seem to have standardized on using the `urfave/cli` package for our go applications. Even though this project is relatively small, it's already missing a few features we get with `urfave/cli`:  environment variable config, integer assertions for the `port` flag, help flag, etc. 

This app also needs to implement the rendering logic @andycmaj mentions in this comment:
https://github.com/quintilesims/orca/pull/31#discussion_r226401904

File: rostrum/Dockerfile (hunk 1 above)
Comment (diemonster) [reply rostrum/Dockerfile]: we definitely need certs

File: rostrum/main.go (hunk 4 above)
Comment (zpatrick) [reply rostrum/main.go]: > As far as rendering goes. The react app is currently designed this way.

I'm not sure exactly what you mean. You will need implement setting `window.initialData` as described in the comment I linked, and then update the react portion to use that as the default state in the redux store.

File: rostrum/src/actions/config.js [hunk 5]
```diff
@@ -21,7 +20,6 @@ export default function getConfig(configUrl) {
         const { items } = response.data;
         const config = {
           ...items,
-          appIsConfigured: true,
```
Comment (tlake) [inline rostrum/src/actions/config.js]: Why are we removing `appIsConfigured` from state, if we're not also removing the action that sets configuration? If we still have any sort of configuration work to do within the app, I think it's still useful to be able to conditionally render based on whether or not configuration is completed. If we don't have any configuration to do within the app, then we should be able to remove this `config.js` and its connections entirely.

File: rostrum/src/actions/namespaces.js [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
 
-
 function namespaceListSuccess(namespaceObjects) {
   return {
     type: NAMESPACE_LIST,
     namespaceObjects,
   };
 }
 
-export function namespaceList(client = K8sClient) {
```
Comment (tlake) [inline rostrum/src/actions/namespaces.js]: These functions will still need to take a `client` param - it just needs to be passed in from the component, and these functions shouldn't supply a default.

`function namespaceList(client) { ... }`

**Comment by zpatrick** [comment]
@nehayward github won't even expand the non-vendored files. can you please remove the vendored code from this pr?

**Pull request closed without merging**